### PR TITLE
multi-slave ST

### PR DESCRIPTION
**Pull request by djosborne**

### CODE REVIEW

**Comment by djosborne** [comment]
I think this ST framework is on the tipping point of warranting a rewrite in Python. Still waiting to see how things develop before I make the jump...

File: test/state.json [hunk 1]
```diff
@@ -0,0 +1 @@
...osmaster","id":"20150807-220927-771756460-5050-1","leader":"master@172.17.0.46:5050","log_dir":"\/var\/log","orphan_tasks":[],"pid":"master@172.17.0.46:5050","slaves":[{"active":true,"attributes":{},"hostname":"slave1.service.consul","id":"20150807-220927-771756460-5050-1-S1","offered_resources":{"cpus":0,"disk":0,"mem":0},"pid":"slave(1)@172.17.0.49:5051","registered_time":1438985371.56728,"reserved_resources":{},"resources":{"cpus":2,"disk":48447,"mem":9707,"ports":"[31000-31100]"},"unreserved_resources":{"cpus":2,"disk":48447,"mem":9707,"ports":"[31000-31100]"},"used_resources":{"cpus":0.1,"disk":0,"mem":128,"ports":"[31016-31016]"}},{"active":true,"attributes":{},"hostname":"slave2.service.consul","id":"20150807-220927-771756460-5050-1-S0","offered_resources":{"cpus":0,"disk":0,"mem":0},"pid":"slave(1)@172.17.0.47:5051","registered_time":1438985371.0069,"reserved_resources":{},"resources":{"cpus":2,"disk":48447,"mem":9707,"ports":"[31000-31100]"},"unreserved_resources":{"cpus":2,"disk":48447,"mem":9707,"ports":"[31000-31100]"},"used_resources":{"cpus":0.1,"disk":0,"mem":128,"ports":"[31047-31047]"}}],"start_time":1438985367.63169,"unregistered_frameworks":[],"version":"0.24.0"}
```
Comment (spikecurtis) [inline test/state.json]: Why have you checked in this file?

Comment (djosborne) [inline test/state.json]: made it past my gitignore somehow. Removing...

**Comment by spikecurtis** [comment]
LGTM, but as mentioned, let's hold off merging

**Pull request merged**